### DISCUSSION TITLE
edgex -sdk

### DISCUSSION THREAD

**Discussion by wangqingpen** [edgexfoundry/community#197] [Q&A]
https://docs.edgexfoundry.org/2.1/getting-started/Ch-GettingStartedSDK-Go/    Using the link above, I have completed that it currently needs to be made into docker form instead of executable, I borrowed from modbus dockerfile file file has the following error, please how to solve it    
![image](https://github.com/edgexfoundry/community/assets/72652818/d87cf994-8450-497c-ae3b-346457e5aafc)

**Comment by MonicaisHer** [comment]
Could you please provide the steps or the specific command you ran that caused the error? By the way, it would be helpful if your log is in English instead of Chinese, as most developers may not be familiar with Chinese. This way, there will be more help available from the community.

**Reply by jiekechoo** [reply]
I can do it. 😄

**Comment by wangqingpen** [comment]
![image](https://github.com/edgexfoundry/community/assets/72652818/2701e9b4-216c-40a5-8685-b0d2f302287a)

The above screenshot is my dockerfile, I use docker in gpio and it works fine, but in the sdk example I create a dockefile and I get the following error
![image](https://github.com/edgexfoundry/community/assets/72652818/54e3e936-1c0f-4a53-9616-d4159430ab17)

**Reply by jiekechoo** [reply]
可以把 go 镜像改成国内的试一下。

**Comment by lenny-goodell** [comment]
Looks like a proxy issue. It is timing out on downloading the SDH module into the running docker build container.

**Comment by bnevis-i** [comment]
Try a `systemctl restart docker` and see if that brings things back to life.

**Comment by jiekechoo** [comment]
@wangqingpen 

1. You can check if  `docker` command is in your `path`.
2. If you use main branch, you need to use `git clone...` to your local location, because when compiling the docker image, you need to use the sha in .git as the image tag;
3. Suggestion 1: Install a Linux (ubuntu or centos) VM in Windows as the compiler;
4. Suggestion 2: Try to use the released version: v2.3, v3.0, v3.1, etc.;

1、你需要确保 `docker` 命令可以在 终端 执行；
2、你如果使用 main branch，需要使用 `git clone ...` 到你的本地，因为编译  docker image 时需要用到 .git 里面的 sha 作为 image tag；
3、建议1：在 Windows 里面安装 Linux （ubuntu或centos）虚拟机作为编译环境，否则可能会出现各种未知问题；
4、建议2：如果不是基于 EdgeX 二次开发，尽量使用发布版本：v2.3, v3.0, v3.1 等；
